controllers/JobController.js

```diff
@@ -164,3 +167,289 @@ exports.addJob = [
+exports.scheduleInterview = [
+  auth,
+  body("jobId", "jobId must not be empty.").notEmpty(),
+  body("candidateId", "candidateId must not be empty.").notEmpty(),
+  body("interviewDate", "interviewDate must not be empty.").notEmpty(),
+  body("interviewTime", "interviewTime must not be empty.").notEmpty(),
+  body("message", "message must not be empty.").notEmpty(),
+  sanitize("*"),
+  (req, res) => {
+    const user = req.user;
+
+    if (user.userType !== constants.userTypes.RECRUITER)
+      return apiResponse.unauthorizedResponse(res, "Invalid Permission.");
+
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return apiResponse.validationErrorWithData(
+        res,
+        "Validation Error.",
+        errors.array()
+      );
+    }
+
+    const { jobId, candidateId, interviewDate, interviewTime, message } =
+      req.body;
+
+    JobModel.findOne({ _id: jobId, postedBy: user.id }, (error, job) => {
+      if (error) {
+        return apiResponse.ErrorResponse(res, "Operation Failed.", error);
+      }
+
+      if (!job) {
+        return apiResponse.ErrorResponse(res, "Job not found.");
+      }
+
+      UserModel.findOne({ _id: candidateId }, (error, user) => {
+        if (error) {
+          return apiResponse.ErrorResponse(res, "Operation Failed.", error);
+        }
+
+        if (!user) {
+          return apiResponse.ErrorResponse(res, "Candidate doesn't exists.");
+        }
+
+        console.log(" jobid & candidateId ", {
+          jobId: mongoose.Types.ObjectId(jobId),
+          candidateId,
+        });
+        JobModel.findOneAndUpdate(
+          {
+            _id: mongoose.Types.ObjectId(jobId),
+            "applicants.candidate": candidateId,
+            "applicants.status": constants.applicationStatus.APPLIED,
+          },
+          {
+            $set: {
+              "applicants.$.status": constants.applicationStatus.INTERVIEWING,
+            },
+          },
+          { new: true },
+          (error, result) => {
+            if (error) {
+              return apiResponse.ErrorResponse(res, "Operation Failed.", error);
+            }
+
+            if (!result)
+              return apiResponse.ErrorResponse(
+                res,
+                "Unable to find candidate."
+              );
+
+            mailer
+              .send(
+                constants.confirmEmails.from,
+                user.email,
+                `Scheduled Interview on ${moment(interviewDate).format(
+                  "DD/mm/yyyy"
+                )} at ${moment(interviewTime).format("hh:mm A")}`,
+                message
+              )
+              .then(() => {
+                return apiResponse.successResponse(res, "Interview scheduled.");
+              })
+              .catch((error) => {
+                if (error) {
+                  return apiResponse.ErrorResponse(
+                    res,
+                    "Operation Failed.",
+                    error
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 10 tabs but found 20 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
										error
```